internal/debug/debug.go

```diff
@@ -0,0 +1,165 @@
+package debug
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type DebugPod struct {
+	Name      string
+	Namespace string
+	NodeName  string
+}
+
+const (
+	interval       = time.Millisecond * 500
+	duration       = time.Second * 5
+	debugNamespace = "openshift-debug"
+)
+
+// New creates debug pod on the given node, puts it in infinite sleep,
+// and returns the DebugPod object. Use the Clean() method to delete it.
+func New(node string) (*DebugPod, error) {
+	errCh := make(chan error)
+	go func() {
+		errCh <- start(node)
+	}()
+	for {
+		select {
+		case err := <-errCh:
+			return nil, err
+		default:
+			return waitForPod()
+		}
+	}
```

Comment:
this seems a bit odd, what is the rationale of not making this "synchronous" (start -> wait)?

Reply:
yes, it's not trivial and took me a couple of tries to figure it out.
It's just that the start func goes to sleep together with the "oc debug... sleep inf.."
So it must be inside a go routine

Reply:
After private discussion - we agreed to look for a different approach,
maybe utilizing an existing privileged pod instead, as this code introduces race condition.